### ISSUE TITLE
TPC-DS query 2 fails assertion

### ISSUE THREAD

**Issue by melanieplageman** [greenplum-db/gpdb#6383]
### Greenplum version or build
 PostgreSQL 9.4beta1 (Greenplum Database 6.0.0-alpha.0+dev.13282.g205f30eedd build dev-oss) on x86_64-apple-darwin16.7.0, compiled by Apple LLVM version 9.0.0 (clang-900.0.39.2), 64-bit compiled on Nov 29 2018 14:42:37 (with assert checking)

### OS version and uname -a
Darwin mplageman 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 21 20:07:39 PDT 2018; root:xnu-3789.73.14~1/RELEASE_X86_64 x86_64

### autoconf options used ( config.status --config )
'--enable-depend' '--with-openssl' '--with-ldap' '--with-libcurl' '--with-python' '--with-perl' '--with-libxml' '--enable-mapreduce' '--prefix=/usr/local/gpdb' '--enable-debug' '--enable-cassert' '--with-libs=/usr/local/opt/openssl/lib /usr/local/opt/libevent/lib' '--with-includes=/usr/local/opt/openssl/include /usr/local/opt/libevent/include' '--with-apr-config=/usr/local/opt/apr/bin/apr-1-config' 'CC=ccache cc' 'CFLAGS=-fno-omit-frame-pointer -O0 -pthread -g -g3' 'CXX=ccache c++' 'CXXFLAGS=-O0 -g -g3 -O0 -g -g3 -O0 -g -g3'

Given that you have an assert build with a demo cluster with the default settings, after running this DDL:
[ddl_tpcds_nonpart_02.txt](https://github.com/greenplum-db/gpdb/files/2635376/ddl_tpcds_nonpart_02.txt)
replace the path to the data with the correct path after extracting the attached data
[smaller_data.zip](https://github.com/greenplum-db/gpdb/files/2635378/smaller_data.zip)

When you run this query:
```
with wscs as
 (select sold_date_sk
        ,sales_price
  from (select ws_sold_date_sk sold_date_sk
              ,ws_ext_sales_price sales_price
        from small_web_sales) x
        union all
       (select cs_sold_date_sk sold_date_sk
              ,cs_ext_sales_price sales_price
        from small_catalog_sales)),
 wswscs as 
 (select d_week_seq,
        sum(case when (d_day_name='Sunday') then sales_price else null end) sun_sales,
        sum(case when (d_day_name='Monday') then sales_price else null end) mon_sales,
        sum(case when (d_day_name='Tuesday') then sales_price else  null end) tue_sales,
        sum(case when (d_day_name='Wednesday') then sales_price else null end) wed_sales,
        sum(case when (d_day_name='Thursday') then sales_price else null end) thu_sales,
        sum(case when (d_day_name='Friday') then sales_price else null end) fri_sales,
        sum(case when (d_day_name='Saturday') then sales_price else null end) sat_sales
 from wscs
     ,date_dim
 where d_date_sk = sold_date_sk
 group by d_week_seq)
 select d_week_seq1
       ,round(sun_sales1/sun_sales2,2)
       ,round(mon_sales1/mon_sales2,2)
       ,round(tue_sales1/tue_sales2,2)
       ,round(wed_sales1/wed_sales2,2)
       ,round(thu_sales1/thu_sales2,2)
       ,round(fri_sales1/fri_sales2,2)
       ,round(sat_sales1/sat_sales2,2)
 from
 (select wswscs.d_week_seq d_week_seq1
        ,sun_sales sun_sales1
        ,mon_sales mon_sales1
        ,tue_sales tue_sales1
        ,wed_sales wed_sales1
        ,thu_sales thu_sales1
        ,fri_sales fri_sales1
        ,sat_sales sat_sales1
  from wswscs,date_dim 
  where date_dim.d_week_seq = wswscs.d_week_seq and
        d_year = 2001) y,
 (select wswscs.d_week_seq d_week_seq2
        ,sun_sales sun_sales2
        ,mon_sales mon_sales2
        ,tue_sales tue_sales2
        ,wed_sales wed_sales2
        ,thu_sales thu_sales2
        ,fri_sales fri_sales2
        ,sat_sales sat_sales2
  from wswscs
      ,date_dim 
  where date_dim.d_week_seq = wswscs.d_week_seq and
        d_year = 2001+1) z
 where d_week_seq1=d_week_seq2-53
 order by d_week_seq1;
```

you should trip this assert:
```
psql:/Users/mplageman/workspace/gp-performance-testing/tpc-ds-nonpart/sql/102.e9.02.sql:58: ERROR:  Unexpected internal error (numeric.c:8062)  (seg0 slice5 127.0.0.1:25432 pid=77830) (numeric.c:8062)
DETAIL:  FailedAssertion("!(var->digits > var->buf)", File: "numeric.c", Line: 8062)
HINT:  Process 77830 will wait for gp_debug_linger=120 seconds before termination.
Note that its locks and other resources will not be released until then.
```
This will happen regardless of which optimizer you plan with.

**Comment by hlinnaka** [comment]
Thanks! Here's a simpler repro:

```
postgres=# select round('0.99897295517141129764', 2);
NOTICE:  round called on 0.99897295517141129764
DETAIL:  FailedAssertion("!(var->digits > var->buf)", File: "numeric.c", Line: 8064)
```

**Comment by hlinnaka** [comment]
Fixed by commit 08542bf27f. Thanks for the report!
